packages/chai-openapi-response-validator/lib/openapi-validator/lib/classes/OpenApi2Spec.js

```diff
@@ -2,10 +2,15 @@ const utils = require('../utils');
 const AbstractOpenApiSpec = require('./AbstractOpenApiSpec');
 const ValidationError = require('./errors/ValidationError');
 
-class OpenApi2Spec extends AbstractOpenApiSpec {
+class OpenApi2Spec extends AbstractOpenApiSpec { 
   findOpenApiPathMatchingPathname(pathname) {
+    const { basePath } = this.spec;
+    if(basePath && !pathname.startsWith(basePath)) {
```

Comment:
if you run our mutation tests, you'll see this isn't needed to pass our tests ;) see the Travis logs https://github.com/RuntimeTools/OpenAPIValidators/pull/160/files#diff-585a46b55b8d80d190a56a2a5ff41a8bR47

So either our tests are insufficient, or this code is unnecesary (you choose, you're the end user!)

Reply:
Good catch. This is necessary, otherwise a spec with `basePath: /test` and path `/responseBody/string` will accept a path of `/responseBody/string`. My new commit "Updated basePath test" now tests exactly this situation. 

Also, I'm not familiar with mutation testing, so I may be missing stuff on that front. Will try to do a bit of research tonight!

Reply:
Thanks for adding the test! Mutation testing is just a way of checking whether you have more code than necessary to pass your tests. We use it here to ensure both that we write no more code than we need to, and that our tests fully cover all the features we want to provide